src/main/java/org/spongepowered/common/profile/SpongeGameProfile.java

```diff
@@ -0,0 +1,254 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.profile;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.mojang.authlib.properties.Property;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.base64.Base64;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.spongepowered.api.data.persistence.DataContainer;
+import org.spongepowered.api.data.persistence.Queries;
+import org.spongepowered.api.profile.GameProfile;
+import org.spongepowered.api.profile.property.ProfileProperty;
+import org.spongepowered.common.util.Constants;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+@SuppressWarnings({"unchecked", "rawtypes"})
+public final class SpongeGameProfile implements GameProfile {
+
+    private static final Gson GSON = new Gson();
+    public static final UUID EMPTY_UUID = new UUID(0, 0);
+
+    public static SpongeGameProfile of(final com.mojang.authlib.GameProfile mcProfile) {
+        final UUID uniqueId = mcProfile.getId() == null ? EMPTY_UUID : mcProfile.getId();
+        final String name = mcProfile.getName();
+        final List<ProfileProperty> properties = (List) ImmutableList.copyOf(mcProfile.getProperties().values());
+        return new SpongeGameProfile(uniqueId, name, properties);
+    }
+
+    public static SpongeGameProfile basicOf(final com.mojang.authlib.GameProfile mcProfile) {
+        final UUID uniqueId = mcProfile.getId() == null ? EMPTY_UUID : mcProfile.getId();
+        final String name = mcProfile.getName();
+        return new SpongeGameProfile(uniqueId, name);
+    }
+
+    public static com.mojang.authlib.GameProfile toMcProfile(final GameProfile profile) {
+        return ((SpongeGameProfile) profile).toMcProfile();
+    }
+
+    public static GameProfile unsignedOf(final GameProfile profile) {
+        if (profile.getProperties().isEmpty()) {
+            return profile;
+        }
+        return new SpongeGameProfile(profile.getUniqueId(), profile.getName().orElse(null),
+                profile.getProperties().stream().map(SpongeGameProfile::withoutSignature).collect(ImmutableList.toImmutableList()));
+    }
+
+    private static String decodeBase64(final String value) {
+        return Base64.decode(Unpooled.wrappedBuffer(value.getBytes(StandardCharsets.UTF_8))).toString(StandardCharsets.UTF_8);
+    }
+
+    private static String encodeBase64(final String value) {
+        return Base64.encode(Unpooled.wrappedBuffer(value.getBytes(StandardCharsets.UTF_8))).toString(StandardCharsets.UTF_8);
```

Comment:
Why are you using Netty here? https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html